### PR TITLE
scrape new AU headless AngularJS table

### PR DESCRIPTION
**Pull request by camjc**
## Summary
Its dynamic so we have to crawl headless :(
But there are Deaths counts now :)
image

### CODE REVIEW

File: src/shared/sources/au/index.js [hunk 1]
```diff
@@ -62,7 +67,63 @@ module.exports = {
...
+      startDate: '2020-04-02',
+      crawl: [
+        {
+          type: 'headless',
+          data: 'table',
+          url: 'https://www.health.gov.au/resources/total-covid-19-cases-and-deaths-by-states-and-territories'
+        }
+      ],
+      scrape ($, date, { assertTotalsAreReasonable, getSchemaKeyFromHeading, normalizeTable }) {
+        const normalizedTable = normalizeTable({ $, tableSelector: '.ng-scope table' })
```
Comment (jzohrab) [inline src/shared/sources/au/index.js:86]: As I mentioned in Slack, I'm not sure what these helper methods do (eg 'schemaKeysByHeadingFragment').  I have a feeling it's good :-) though it does make the code more opaque.

File: src/shared/sources/au/index.js [hunk 2]
```diff
@@ -62,7 +67,63 @@ module.exports = {
...
+            deaths: parse.number(stateData.deaths)
+          })
+        })
+
+        const summedData = transform.sumData(states)
+        states.push(summedData)
+
+        const indexForCases = dataKeysByColumnIndex.findIndex(key => key === 'cases')
+        const casesFromTotalRow = parse.number(
+          normalizedTable.find((row) => row.some(column => column === totalLabel))[indexForCases]
```
Comment (jzohrab) [inline src/shared/sources/au/index.js:126]: This line is very tricky!  What's going on in it?

Comment (camjc) [reply src/shared/sources/au/index.js:126]: Go find the row that says "Australia" and grab the case number from it.
We later use it to compare to the calculated total as a sanity check.

File: src/shared/sources/au/index.js (hunk 1 above)
Comment (camjc) [reply src/shared/sources/au/index.js:86]: Lets workshop some renames for these utils once you have a read of the JSdoc and get a clearer picture of what they're for? Definitely want to make the naming clear for folks :)

File: src/shared/sources/au/index.js [hunk 3]
```diff
@@ -62,7 +67,63 @@ module.exports = {
...
+          type: 'headless',
+          data: 'table',
+          url: 'https://www.health.gov.au/resources/total-covid-19-cases-and-deaths-by-states-and-territories'
+        }
+      ],
+      scrape ($, date, { assertTotalsAreReasonable, getSchemaKeyFromHeading, normalizeTable }) {
+        const normalizedTable = normalizeTable({ $, tableSelector: '.ng-scope table' })
+
+        const headingRowIndex = 0
+        console.log(normalizedTable)
```
Comment (camjc) [inline src/shared/sources/au/index.js:89]:
```suggestion

```

**Comment by camjc** [comment]
Blocked by headless scrapes failing in CI (looks like its trying to run head-ful)

File: src/shared/sources/au/index.js (hunk 2 above)
Comment (ryanblock) [reply src/shared/sources/au/index.js:124]: Can we decompose this please?

**Pull request merged**